codex/rest/api.nim

```diff
@@ -218,27 +218,32 @@ proc initRestApi*(node: CodexNodeRef, conf: CodexConf): RestRouter =
   router.rawApi(
     MethodPost,
     "/api/codex/v1/upload") do (
+      chunk: Option[uint]
     ) -> RestApiResponse:
-      ## Upload a file in a streamming manner
+      ## Upload a file in a streamming manner,
+      ## split it into blocks of `chunk` bytes,
+      ## and save to the node's BlockStore
       ##
 
       trace "Handling file upload"
-      var bodyReader = request.getBodyReader()
+      let bodyReader = request.getBodyReader()
       if bodyReader.isErr():
-        return RestApiResponse.error(Http500)
+        return RestApiResponse.error(Http500, bodyReader.error)
 
-      # Attempt to handle `Expect` header
-      # some clients (curl), wait 1000ms
-      # before giving up
-      #
+      # Attempt to handle `Expect` header some clients (curl),
+      # wait 1000ms before giving up
       await request.handleExpect()
 
-      let
-        reader = bodyReader.get()
+      let bodyStream = AsyncStreamWrapper.new(reader = AsyncStreamReader(bodyReader.get))
```

Comment:
is there a possibility that AsyncStreamWrapper.new will raise an exception? I ask that because in prior code, this operation was inside `try`, and now I moved it our in order to call bodyStream.closeImpl in `finally`. 

If we need to catch exceptions in AsyncStreamWrapper.new too, I will write a bit more complex code.

Reply:
Yes, it could because `new` will call `initStream`, which does some additional work that could raise. I would move the assignement into the `try` but leave the declaration as a `var bodyStream`... Not ideal, but not much we can do.

Reply:
We should move `AsyncStreamWrapper.new` inside the try...